rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp

```diff
@@ -1014,21 +1032,41 @@ bool TypeSupport<MembersType>::deserialize(
   assert(data);
   assert(payload);
 
-  auto buffer = static_cast<eprosima::fastcdr::FastBuffer *>(data);
-  if (!buffer->reserve(payload->length)) {
-    return false;
+  auto ser_data = static_cast<SerializedData *>(data);
```

Comment:
i think this function also could be strongly typed.

Reply:
Not possible, as it is the implementation of Fast-RTPS TopicDataType interface, declared in [this line](https://github.com/eProsima/Fast-RTPS/blob/fb91928914f7f8e47d0ef10bae26b207524b75ad/include/fastrtps/TopicDataType.h#L60)